cis194/week2/wei-ning/src/Exercise5.hs

```diff
@@ -0,0 +1,24 @@
+-- perl -wnl -E '/Error\s+(\d+)/ or next; say $_ if $1 >= 50' /var/tmp/sut/prt.txt
+
+module Exercise5
+  ( whatWentWrong
+  ) where
+
+import           Log
+import           Prelude hiding (Error)
+
+whatWentWrong :: [LogMessage] -> [String]
+whatWentWrong = filterMessages (Error 50)
+
+filterMessages :: MessageType -> [LogMessage] -> [String]
+filterMessages (Error lv) =
+  map extractMessage . filter isCriticalError
+  where
+    isCriticalError (LogMessage (Error lv') _ _) | lv' >= lv = True
```

Comment:
I don’t think we need guard here

`isCriticalError (LogMessage (Error lv') _ _) = lv' >= lv`

Reply:
yep totally right